shell/platform/darwin/ios/framework/Source/FlutterPlatformViews.mm

```diff
@@ -127,6 +125,28 @@
 
   UIView* view = views_[view_id];
   [view setFrame:rect];
+  composition_order_.push_back(view_id);
+}
+
+void FlutterPlatformViewsController::Present() {
+  if (composition_order_ == active_composition_order_) {
+    composition_order_.clear();
+    return;
+  }
+  UIView* flutter_view = flutter_view_.get();
+
+  for (UIView* sub_view in [flutter_view subviews]) {
+    [sub_view removeFromSuperview];
```

Comment:
Adding and removing subviews will trigger unnecessary layout passes in their view hierarchies. Instead, please iterate over the views in composition order and call `bringSubviewToFront:`

Reply:
So something like, add the view if it has not superview ortherwise just bring it to the front before iterating to the next one in composition order.

Reply:
Boy I'm glad you are reviewing this :)

As discussed for now I'm just leaving a TODO, as right now it's a little hard to test, and this shouldn't happen very often.